<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>KnightsTourClass</class>
 <widget class="QMainWindow" name="KnightsTourClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>896</width>
    <height>798</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>KnightsTour</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout_3" rowstretch="0,1,0,0">
    <item row="0" column="0" rowspan="3">
     <widget class="QGroupBox" name="groupboxMoves">
      <property name="maximumSize">
       <size>
        <width>200</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="title">
       <string>Трасса ходов</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_2">
       <item row="0" column="0">
        <widget class="QListWidget" name="listMoves">
         <property name="statusTip">
          <string>Список ходов</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="0" column="1">
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <widget class="QLabel" name="labelFieldSize">
        <property name="text">
         <string>Размер поля. Ширина:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="spinboxWidth">
        <property name="statusTip">
         <string>Ширина поля от 3 до 200 клеток</string>
        </property>
        <property name="minimum">
         <number>3</number>
        </property>
        <property name="maximum">
         <number>200</number>
        </property>
        <property name="value">
         <number>8</number>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="labelX">
        <property name="text">
         <string>Высота:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="spinBoxHeight">
        <property name="statusTip">
         <string>Высота поля от 3 до 200 клеток</string>
        </property>
        <property name="minimum">
         <number>3</number>
        </property>
        <property name="maximum">
         <number>200</number>
        </property>
        <property name="value">
         <number>8</number>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="buttonSetFieldSize">
        <property name="statusTip">
         <string>Задать размер поля</string>
        </property>
        <property name="text">
         <string>Начать заново</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="0" column="2" colspan="2">
     <spacer name="horizontalSpacer_3">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>380</width>
        <height>20</height>
       </size>
      </property>
     </spacer>
    </item>
    <item row="1" column="1" colspan="3">
     <widget class="FieldViewWidget" name="fieldWidget" native="true">
      <property name="statusTip">
       <string>Поле. Щёлкните по клетке, чтобы задать начальное положение фигуры</string>
      </property>
     </widget>
    </item>
    <item row="2" column="1" colspan="2">
     <layout class="QGridLayout" name="gridLayout" columnstretch="0,0,0,0,0">
      <item row="0" column="0">
       <widget class="QRadioButton" name="radioManual">
        <property name="statusTip">
         <string>Пошаговое перемещение фигуры</string>
        </property>
        <property name="text">
         <string>Вручную</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="0" column="3">
       <widget class="QPushButton" name="buttonBack">
        <property name="statusTip">
         <string>Шаг назад</string>
        </property>
        <property name="text">
         <string>&lt;&lt;</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QRadioButton" name="radioAuto">
        <property name="statusTip">
         <string>Автоматическое перемещение фигуры</string>
        </property>
        <property name="text">
         <string>Автоматически с задержкой</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QSpinBox" name="spinboxDelay">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="statusTip">
         <string>Задержка после каждого хода</string>
        </property>
        <property name="suffix">
         <string> мс</string>
        </property>
        <property name="maximum">
         <number>10000</number>
        </property>
        <property name="singleStep">
         <number>125</number>
        </property>
        <property name="value">
         <number>125</number>
        </property>
       </widget>
      </item>
      <item row="1" column="3">
       <widget class="QPushButton" name="buttonGo">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="statusTip">
         <string>Начать автоматическое перемещение</string>
        </property>
        <property name="text">
         <string>Начать</string>
        </property>
       </widget>
      </item>
      <item row="1" column="4">
       <widget class="QPushButton" name="buttonStop">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="statusTip">
         <string>Остановить фигуру</string>
        </property>
        <property name="text">
         <string>Стоп</string>
        </property>
       </widget>
      </item>
      <item row="0" column="4">
       <widget class="QPushButton" name="buttonForward">
        <property name="statusTip">
         <string>Шаг вперёд</string>
        </property>
        <property name="text">
         <string>&gt;&gt;</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="2" column="3">
     <spacer name="horizontalSpacer_2">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>260</width>
        <height>28</height>
       </size>
      </property>
     </spacer>
    </item>
    <item row="3" column="0" colspan="4">
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>578</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QPushButton" name="buttonExit">
        <property name="maximumSize">
         <size>
          <width>75</width>
          <height>23</height>
         </size>
        </property>
        <property name="statusTip">
         <string>Выход из программы</string>
        </property>
        <property name="text">
         <string>Выход</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>896</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>О программе</string>
    </property>
    <addaction name="actionAbout"/>
    <addaction name="actionAbout_Qt"/>
   </widget>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionAbout">
   <property name="text">
    <string>О программе</string>
   </property>
  </action>
  <action name="actionAbout_Qt">
   <property name="text">
    <string>О Qt</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>FieldViewWidget</class>
   <extends>QWidget</extends>
   <header>fieldviewwidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="knightstour.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>radioManual</sender>
   <signal>toggled(bool)</signal>
   <receiver>buttonBack</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>327</x>
     <y>704</y>
    </hint>
    <hint type="destinationlabel">
     <x>479</x>
     <y>707</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>radioManual</sender>
   <signal>toggled(bool)</signal>
   <receiver>buttonForward</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>327</x>
     <y>704</y>
    </hint>
    <hint type="destinationlabel">
     <x>560</x>
     <y>707</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>radioAuto</sender>
   <signal>toggled(bool)</signal>
   <receiver>spinboxDelay</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>327</x>
     <y>733</y>
    </hint>
    <hint type="destinationlabel">
     <x>398</x>
     <y>734</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>radioAuto</sender>
   <signal>toggled(bool)</signal>
   <receiver>buttonGo</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>327</x>
     <y>733</y>
    </hint>
    <hint type="destinationlabel">
     <x>479</x>
     <y>736</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>radioAuto</sender>
   <signal>toggled(bool)</signal>
   <receiver>buttonStop</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>327</x>
     <y>733</y>
    </hint>
    <hint type="destinationlabel">
     <x>560</x>
     <y>736</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinboxWidth</sender>
   <signal>valueChanged(int)</signal>
   <receiver>buttonSetFieldSize</receiver>
   <slot>click()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>290</x>
     <y>55</y>
    </hint>
    <hint type="destinationlabel">
     <x>444</x>
     <y>59</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinBoxHeight</sender>
   <signal>valueChanged(int)</signal>
   <receiver>buttonSetFieldSize</receiver>
   <slot>click()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>390</x>
     <y>52</y>
    </hint>
    <hint type="destinationlabel">
     <x>470</x>
     <y>49</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
